### PR TITLE
[AMDGPU] Port AMDGPURewriteUndefForPHI to new pass manager

### PR DESCRIPTION
**Pull request by jwanggit86**
This patch ports the AMDGPURewriteUndefForPHI pass to the new pass manager. With this, the pass is supported under both the legacy and the new pass managers.

### CODE REVIEW

File: llvm/lib/Target/AMDGPU/AMDGPURewriteUndefForPHI.cpp [hunk 1]
```diff
@@ -177,6 +177,20 @@ bool AMDGPURewriteUndefForPHI::runOnFunction(Function &F) {
...
+  if (Changed) {
+    PreservedAnalyses PA;
+    PA.preserveSet<CFGAnalyses>();
+    return PA;
+  }
+
+  return PreservedAnalyses::all();
+}
+
 FunctionPass *llvm::createAMDGPURewriteUndefForPHIPass() {
```
Comment (arsenm) [inline llvm/lib/Target/AMDGPU/AMDGPURewriteUndefForPHI.cpp:194]: Should rename all the legacy pass bits to end with "LegacyPass"

Comment (jwanggit86) [reply llvm/lib/Target/AMDGPU/AMDGPURewriteUndefForPHI.cpp:194]: Should the class "AMDGPURewriteUndefForPHI" be renamed "AMDGPURewriteUndefForPHILegacy" or "AMDGPURewriteUndefForPHILegacyPass"? If the latter, we will end up with a function named "initializeAMDGPURewriteUndefForPHILegacyPassPass".

Comment (arsenm) [reply llvm/lib/Target/AMDGPU/AMDGPURewriteUndefForPHI.cpp:194]: AMDGPURewriteUndefForPHILegacy, some places have always been stuck with "PassPass"

File: llvm/lib/Target/AMDGPU/AMDGPURewriteUndefForPHI.cpp [hunk 2]
```diff
@@ -170,13 +170,27 @@ bool rewritePHIs(Function &F, UniformityInfo &UA, DominatorTree *DT) {
...
   DominatorTree *DT = &getAnalysis<DominatorTreeWrapperPass>().getDomTree();
   return rewritePHIs(F, UA, DT);
 }
 
-FunctionPass *llvm::createAMDGPURewriteUndefForPHIPass() {
-  return new AMDGPURewriteUndefForPHI();
+PreservedAnalyses
+AMDGPURewriteUndefForPHIPass::run(Function &F, FunctionAnalysisManager &AM) {
+  UniformityInfo &UA = AM.getResult<UniformityInfoAnalysis>(F);
+  DominatorTree *DT = &AM.getResult<DominatorTreeAnalysis>(F);
```
Comment (arsenm) [inline llvm/lib/Target/AMDGPU/AMDGPURewriteUndefForPHI.cpp:183]: Why address of here?

**Review by arsenm — APPROVED** [review]

**Pull request merged**